infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationEventListener.java

```diff
@@ -81,6 +96,30 @@ public void onNext(CorfuStreamEntries results) {
         }
     }
 
+    /**
+     * Read the event table to process any events written when LR was not running.
+     */
+    private void processPendingRequests() {
+        List<CorfuStoreEntry<ReplicationEventInfoKey, ReplicationEvent, Message>> pendingEvents =
+            discoveryService.getSessionManager().getMetadataManager().getReplicationEvents();
+
+        for (CorfuStoreEntry event : pendingEvents) {
+            if (((ReplicationEvent)event.getPayload()).getType().equals(
+                    ReplicationEventType.UPGRADE_COMPLETION_FORCE_SNAPSHOT_SYNC)) {
+                triggerForcedSnapshotSyncForAllSessions((ReplicationEvent)event.getPayload());
+            }
+        }
+    }
```

Comment:
nit: This can be a todo but maybe we can abstract the logic to process the events to reduce overlap with onNext() incase we add more in the future.

Reply:
good point.  I'll add a TODO for it.